### PR TITLE
feat: ability to define colors for a game

### PR DESCRIPTION
**Pull request by kelsin**
This uses the colors names from 18xx-maker in the game definition, which WILL pass through if not defined. This means if we don't define colors the svg will just use `red` or `green` and the like. However now you can call `register_colors` to define the colors for this game in the game file. I did it for 1889 and 18Chesapeake.

For 1889 and 18Chesapeake I used the colors currently used.
For 1846 and 1830 I used the GMT color scheme from 18xx-maker.
For 1817 I used every Rails on Board token color (one for each of the 20), we will probably change this if you want to use AAG's colors of course.

At least now changing that is just on your side 👍

### CODE REVIEW

File: lib/engine/config/game/g_1889.rb [hunk 1]
```diff
@@ -231,7 +231,7 @@ module G1889
       "name": "Mitsubishi Ferry",
       "value": 30,
       "revenue": 5,
-      "desc": "Player owner may place the port tile on a coastal town (B11, G10, I12, or J9) without a tile on it already, outside of the operating rounds of a corporation controlled by another player. The player need not control a corporation or have connectivity to the placed tile from one of their corporations. This does not close the company.",
+      "desc": "Player owner may place the port tile on a coastal town (B11, G10, I12, or J9) without a tile on it already, outside of the operating rounds of a corporation controlled by another player. The player need not control a corporation or have connectivity to the placed tile from one of their corporations. This does not close the corporation.",
```
Comment (kelsin) [inline lib/engine/config/game/g_1889.rb:234]: I fixed my description but looks like you missed one `company` to `corporation` so fixing.

File: lib/engine/config/game/g_18_chesapeake.rb [hunk 2]
```diff
@@ -691,7 +697,8 @@ module G18Chesapeake
       "tiles": [
         "yellow",
         "green",
-        "brown"
+        "brown",
+        "gray"
```
Comment (kelsin) [inline lib/engine/config/game/g_18_chesapeake.rb:701]: This was a bug pointed out in chat, doesn't look fixed yet in the code so added here.

File: lib/engine/game/base.rb [hunk 3]
```diff
@@ -79,6 +79,14 @@ def self.title
         name.split('::').last.slice(1..-1)
       end
 
+      def self.register_colors(colors)
+        colors.default_proc = proc do |_, key|
+          key
+        end
```
Comment (kelsin) [inline lib/engine/game/base.rb:85]: This makes sure that when you don't define a certain color it just passes the color name through (which works fine in SVG)

File: lib/engine/config/game/g_1830.rb [hunk 4]
```diff
@@ -343,33 +343,33 @@ module G1830
...
       "coordinates": "I15",
-      "color": "#0189d1"
+      "color": "blue"
     },
     {
       "float_percent": 60,
       "sym": "C&O",
       "name": "Chesapeake & Ohio Railroad",
-      "logo": "1830/CO",
+      "logo": "1830/C&O",
```
Comment (tobymao) [inline lib/engine/config/game/g_1830.rb:365]: we can''t use & in logos anymore, i think nginx strips them out or something

File: lib/engine/config/game/g_1846.rb [hunk 5]
```diff
@@ -237,34 +237,36 @@ module G1846
...
       "coordinates": "D20",
-      "color": "#37383a"
+      "color": "black"
     },
     {
       "float_percent": 20,
       "sym": "B&O",
       "name": "Baltimore & Ohio",
-      "logo": "1846/BO",
+      "logo": "1846/B&O",
```
Comment (tobymao) [inline lib/engine/config/game/g_1846.rb:246]: remove all &

File: lib/engine/config/game/g_1889.rb (hunk 1 above)
Comment (tobymao) [reply lib/engine/config/game/g_1889.rb:234]: no this references the company as corporations don't close

File: lib/engine/config/game/g_1830.rb (hunk 4 above)
Comment (kelsin) [reply lib/engine/config/game/g_1830.rb]: Done.

File: lib/engine/config/game/g_1889.rb (hunk 1 above)
Comment (kelsin) [reply lib/engine/config/game/g_1889.rb]: Yep.... dumb. Fixed.

**Pull request merged**